pkg/utils/common/common.go

```diff
@@ -239,6 +240,51 @@ func RealtimePrintCommandOutput(cmd *exec.Cmd, logFile string) error {
 	return nil
 }
 
+// ClusterObject2Map convert ClusterObjectReference to a readable map
+func ClusterObject2Map(refs ...common.ClusterObjectReference) map[string]string {
+	clusterResourceRefTmpl := "Cluster: %s | Namespace: %s | GVK: %s/%s | Name: %s"
+	objs := make(map[string]string, len(refs))
+	for _, r := range refs {
+		if r.Cluster == "" {
+			r.Cluster = "local"
+		}
+		objs[r.Name] = fmt.Sprintf(clusterResourceRefTmpl, r.Cluster, r.Namespace, r.APIVersion, r.ResourceVersion, r.Name)
+	}
+	return objs
+}
+
+// AskToChooseOneAppliedResource will ask users to select one applied resource of the application if more than one
+// resources is a map for component to applied resources
+// return the selected ClusterObjectReference
+func AskToChooseOneAppliedResource(resources []common.ClusterObjectReference) (*common.ClusterObjectReference, error) {
+	if len(resources) == 0 {
+		return nil, fmt.Errorf("no applied resources exist in the application")
```

Comment:
What's an `applied resource`?

Reply:
It's generally a combination of {cluster, gvk, namespace, name}